### PR TITLE
Authorised user idle timeout

### PR DESCRIPTION
**Pull request by DylanSpong**
Models.py imported User model twice. This has been tested on two machines, Both mine and Vidul's machine with his updated sign-up/sign-in task.

### CODE REVIEW

**Comment by DylanSpong** [comment]
Demo link here - 
https://youtu.be/hcFATomd9WU

File: home/idle.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ 
+        # If idle time exceeds session timeout, log out the user
+        if idle_time > settings.SESSION_COOKIE_AGE:
+            logout(request)
+            return redirect('login')
+ 
+        # Update the session with the current activity timestamp
+        request.session['last_activity'] = current_time.timestamp()
+        return self.get_response(request)
+
```
Comment (s-bonnor) [inline home/idle.py:34]: Can you justify the reason for using a full middleware for something that can just be done by setting the max_age value on the session cookie? or what advantages this has over doing so?

Comment (DylanSpong) [reply home/idle.py:34]: This is a basic middleware that will logout the user after being inactive for 10 minutes. I believe having the full middleware will make it easier for new juniors next trimester to expand upon session management as they will be able to understand the fields that have been used and how they apply to the feature. I was planning to implement storage of session IDs to be reinstated upon login; however, time got the better of me so I can continue with this next trimester. I was also unaware that the max_age for session cookies was the equivalent of what I have produced.

**Review by DylanSpong — COMMENTED** [review]
Changes have been made to files needing review to stay consistent with current merges.

**Review by s-bonnor — APPROVED** [review]
Looks fine

**Pull request merged**